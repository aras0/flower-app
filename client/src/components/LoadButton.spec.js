import { createLocalVue, shallowMount } from '@vue/test-utils';
import Button from './LoadButton.vue';

const localVue = createLocalVue();

describe('load-button', () => {
  it('renders correctly with default options', () => {
    const wrapper = shallowMount(Button, { localVue });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correctly with custom options', () => {
    const wrapper = shallowMount(Button, {
      localVue,
      propsData: {
        type: 'secondary',
        disabled: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('validates button type', () => {
    const errorMock = jest.spyOn(global.console, 'error').mockImplementation();
    shallowMount(Button, {
      localVue,
      propsData: { type: 'invalid' },
    });
    expect(errorMock).toHaveBeenCalled();
    errorMock.mockRestore();
  });

  it('displays the loading status with a custom slot', () => {
    const wrapper = shallowMount(Button, {
      localVue,
      propsData: { loading: true },
      slots: { loading: '<p>Custom slot</p>' },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
