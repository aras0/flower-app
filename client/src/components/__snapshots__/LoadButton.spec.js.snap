// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load-button displays the loading status with a custom slot 1`] = `
<button
  class="button-primary"
  disabled="disabled"
>
  <p>
    Custom slot
  </p>
</button>
`;

exports[`load-button renders correctly with custom options 1`] = `
<button
  class="button-secondary"
  disabled="disabled"
/>
`;

exports[`load-button renders correctly with default options 1`] = `
<button
  class="button-primary"
/>
`;
